crates/bevy_reflect/src/impls/smol_str.rs

```diff
@@ -1,8 +1,14 @@
-use crate::std_traits::ReflectDefault;
 use crate::{self as bevy_reflect};
+use crate::{std_traits::ReflectDefault, ReflectDeserialize};
 use bevy_reflect_derive::impl_reflect_value;
 
-impl_reflect_value!(::smol_str::SmolStr(Debug, Hash, PartialEq, Default));
+impl_reflect_value!(::smol_str::SmolStr(
+    Debug,
+    Hash,
+    PartialEq,
+    Default,
+    Deserialize,
```

Comment:
```suggestion
    Serialize,
    Deserialize,
```

Reply:
Oh yeah that's perfectly reasonable, don't know how I missed that!